test/fastly/client_test.rb

```diff
@@ -133,4 +133,54 @@
 
     end
   end
+
+  describe 'retry when something goes wrong' do
```

Comment:
Although being fault tolerant is generally a good thing, is there a reason why we're implementing retries into the API client and could this enhancement support an opt-in or opt-out configuration option.

If we implement this, then it would likely be considered a major semver breaking change, and some users might not want the retry behaviour by default (or at all). So maybe making this optional and defaulting to the behaviour not being enabled so we don't have to do a major version bump unnecessarily.

Reply:
I'll think about if there is a simple way to make it optional.

One reason for limiting the retries to just the case of response error messages matching the specific strings is to make this as light as possible and to really only retry when it is desirable to do so. So you might concede a configuration option isn't necessary.